src/helpers/distance.ts

```diff
@@ -0,0 +1,38 @@
+import { GroundPlace } from 'src/types';
+
+/**
+ * @description This function uses the ‘haversine’ formula to calculate the great-circle distance between two points.
+ *
+ * That is the shortest distance over the earth’s surface.
+ *
+ * Original calculations from 'https://www.movable-type.co.uk/scripts/latlong.html'.
+ * @param {GroundPlace} firstPlace - The first place to check, it could be a StopGroup or a StopCluster.
+ * @param {GroundPlace} secondPlace - The second place to check, it also could be a StopGroup or a StopCluster.
+ */
+const distanceBetweenTwoPlaceInKms = (firstPlace: GroundPlace, secondPlace: GroundPlace): number => {
+  const { latitude: firstPlaceLat, longitude: firstPlaceLong } = firstPlace;
+  const { latitude: secondPlaceLat, longitude: secondPlaceLong } = secondPlace;
+
+  if (!firstPlaceLat || !firstPlaceLat || !secondPlaceLat || !secondPlaceLong) {
```

Comment:
watch out you have firstPlaceLat twice, missing firstPlaceLong

Reply:
Yes, miss this one! Thanks!